### PR TITLE
added a config for adapter

### PR DESCRIPTION
**Pull request by pk82869**
adapter is now configurable using a json file, defaults to 'file' mode with rbac data.

### CODE REVIEW

File: server/adapter.go [hunk 1]
```diff
@@ -56,3 +59,31 @@ func newAdapter(in *pb.NewAdapterRequest) (persist.Adapter, error) {
...
+	decoder := json.NewDecoder(configFile)
+	config := Config{}
+	decoder.Decode(&config)
+	return config
+}
+
+type Config struct {
+	Driver string
+	Connection string
+}
```
Comment (hsluoyz) [inline server/adapter.go]: Add a new line here.

File: server/adapter.go [hunk 2]
```diff
@@ -30,8 +33,8 @@ var errDriverName = errors.New("currently supported DriverName: file | mysql | p
 
 func newAdapter(in *pb.NewAdapterRequest) (persist.Adapter, error) {
 	var a persist.Adapter
+	in = checkAdapter(in)
 	supportDriverNames := [...]string{"file", "mysql", "postgres", "mssql"}
-
```
Comment (hsluoyz) [inline server/adapter.go:34]: Leave the new line here.

File: server/adapter.go [hunk 3]
```diff
@@ -56,3 +59,31 @@ func newAdapter(in *pb.NewAdapterRequest) (persist.Adapter, error) {
 
 	return a, nil
 }
+
+func checkAdapter(in *pb.NewAdapterRequest) *pb.NewAdapterRequest {
```
Comment (hsluoyz) [inline server/adapter.go]: Rename to checkLocalConfig

**Review by hsluoyz — CHANGES_REQUESTED** [review]
Please fix.

**Comment by pk82869** [comment]
> Please fix.

Done.

**Comment by pk82869** [comment]
Should I add a similar functionality for enforcer as well? As in, an option to read from config in case nothing is specified.

**Comment by hsluoyz** [comment]
OK. But it should not be default. If the config file name is abc.conf, make it first abc.conf.bak in repo.

**Pull request merged**